blocks/header-nav-chain-block/chains/header-nav-chain-block/default.jsx

```diff
@@ -170,6 +154,20 @@ const Nav = (props) => {
 
   const [onScrollDebounced] = useDebouncedCallback(onScrollEvent, 100);
 
+  // istanbul ignore next
+  useEffect(() => {
+    const handleEscKey = (event) => {
+      if (event.keyCode === 27) {
+        closeNavigation();
+      }
+    };
+
+    window.addEventListener('keydown', handleEscKey, true);
+    return () => {
+      window.removeEventListener('keydown', handleEscKey);
+    };
+  }, []);
+
   useEffect(() => {
     const mastHead = document.querySelector('.masthead-block-container .masthead-block-logo');
```

Comment:
Wondering if this should be accessed via a ref at this point?  Don't think however its urgent right now

Reply:
Yeah I completely agree 🙂 We can definitely address that in a follow-up ticket at some point. I think there's some additional refactoring and clean-up that we could benefit from once we have some time.